### PR TITLE
Add special variables

### PR DESCRIPTION
**Pull request by jacwah**
This PR builds on #83. The tests were failing because pathwatcher was failing to install. I merged it with master were the pathwatcher dependency has been removed.

> Adds variables like ${/filename} that expand into special values, much like __filename in node. The prefix character is designed to not conflict with normal variable names if/when they are supported.

### CODE REVIEW

File: lib/snippet.coffee [hunk 1]
```diff
@@ -1,5 +1,6 @@
 _ = require 'underscore-plus'
-{Range} = require 'atom'
+{Range}  = require 'atom'
```
Comment (50Wliu) [inline lib/snippet.coffee]: Can you revert this back to the single-space variant?

File: lib/variable.coffee [hunk 2]
```diff
@@ -0,0 +1,60 @@
+
```
Comment (50Wliu) [inline lib/variable.coffee]: Extra blank line

File: lib/variable.coffee [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+fs     = require 'fs'
+path   = require 'path'
+moment = require 'moment'
```
Comment (50Wliu) [inline lib/variable.coffee]: Also don't align these

File: lib/variable.coffee [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+lineNumMagicStr = '~l#N~'
+
+exports.getValue = (varName) ->
+  unixify  = (path) -> path.replace(/\\/g, '/')
+
+  editor   = atom.workspace.getActiveTextEditor()
+  if not varName or not editor then return ''
+
+  filePath = unixify(editor.getPath())
+  project  = atom.project
```
Comment (50Wliu) [inline lib/variable.coffee]: Ditto!

**Comment by yniknafs** [comment]
Hey sorry i'm not great with git. Is it possible for me to install this before its "merged"?

**Comment by jacwah** [comment]
@yniknafs I'm not sure how you'd install it to your regular Atom, maybe someone else can chime in. 

Something you can do is to install it as a development package and run Atom in development mode. To do this, first install my fork of snippets as a dev package: `apm develop jacwah/snippets`. Then find the path APM installed the package in with `apm ls`. Go to the specified folder and switch to the variables branch with `git checkout variables`.

Now if you launch Atom in dev mode you should have the new features available! Either run `atom --dev` or go to **View>Developer>Open In Dev Mode...** in an open Atom window.

**Comment by farism** [comment]
Any word on getting this merged?

File: lib/snippet-body.pegjs [hunk 5]
```diff
@@ -1,6 +1,7 @@
-bodyContent = content:(tabStop / bodyContentText)* { return content; }
+
```
Comment (ErikBean) [inline lib/snippet-body.pegjs]: Did you mean to add this extra blank line?

**Comment by sbquinlan** [comment]
@nathansobo any chance of getting this merged?

**Comment by IanVS** [comment]
It's a shame this has been stalled for the past year.  I would really love to have this feature, is there anything I could do to help get it across the finish line?  It looks like it needs a rebase.  It's a shame it seems to have been stalled on style problems (which appear to have been addressed).

**Comment by brigand** [comment]
I tried @jacwah 's instructions, but it gave "Request for package information failed: Not Found". Is there some way to install this branch?

**Comment by jacwah** [comment]
@brigand This branch is probably very out of date with current Atom since it's nearly two years old. 

I'm not using Atom myself anymore. Anyone who is interested, feel free to fork and update this branch if they wish.

**Pull request closed without merging**